Microsoft.Research/AbstractInterpretation/Abstract Domains/Utils.cs

```diff
@@ -866,6 +866,12 @@ virtual public Out VisitGreaterThan_Un(Expression left, Expression right, Expres
             Contract.Requires(left != null);
             Contract.Requires(right != null);
 
+            // the Roslyn compiler uses cgt.un for comparing references to null; while producing correct
+            // results at runtime, it doesn't match implicit non-null assumptions made by the static checker,
+            // so we detect it here and convert it to an old-style equality comparison
```

Comment:
:thought_balloon: Consider rewording this paragraph. I don't feel that the references to "Roslyn" and "old-style equality comparison" help the reader understand why this code is needed. In fact, even if Roslyn *didn't* use this style of null check, it would still make since to include this. The following comes to mind.

> Since there is no `cne` instruction, ECMA-335 §III.1.5 makes a note that `cgt.un` may be used instead for the specific case where the right-hand-side is null. For all integer inputs to the `cgt.un` instruction, if the right side is null or zero, we treat the instruction as a "not equal" instruction for improved results from the static checker.